apps/chat_server/lib/chat_server/contexts/rooms/update_state.ex

```diff
@@ -0,0 +1,38 @@
+defmodule ChatServer.UpdateState do
+  require Logger
+
+  alias ChatServer.Schema
+
+  defmodule State do
+    @derive {Poison.Encoder, only: [room: [:id, :slug, :name]]}
+
+    defstruct [:room]
+  end
+
+  def call(params \\ %{}, _user, state) do
```

Comment:
Unless there's a use case where we want the funtion to default to an empty map and still execute, maybe this should just be `def call(params, _user, state) do`?